src/wasm/wasm-s-parser.cpp

```diff
@@ -902,67 +672,11 @@ Expression* SExpressionWasmBuilder::makeBinary(Element& s, BinaryOp op, Type typ
 }
 
 
-Expression* SExpressionWasmBuilder::makeUnary(Element& s, UnaryOp op, Type type) {
+Expression* SExpressionWasmBuilder::makeUnary(Element& s, UnaryOp op) {
   auto ret = allocator.alloc<Unary>();
   ret->op = op;
   ret->value = parseExpression(s[1]);
   ret->finalize();
-  // type is the reported type, e.g. i64.ctz reports i64 (but has a return type of i32, in this case)
-  // verify the reported type is correct
-  switch (op) {
-    case EqZInt32:
-    case NegFloat32:
-    case AbsFloat32:
-    case CeilFloat32:
-    case FloorFloat32:
-    case TruncFloat32:
-    case NearestFloat32:
-    case SqrtFloat32:
-    case ClzInt32:
-    case CtzInt32:
-    case PopcntInt32:
-    case EqZInt64:
-    case NegFloat64:
-    case AbsFloat64:
-    case CeilFloat64:
-    case FloorFloat64:
-    case TruncFloat64:
-    case NearestFloat64:
-    case SqrtFloat64:
-    case ClzInt64:
-    case CtzInt64:
-    case PopcntInt64: {
-      if (ret->value->type != unreachable && type != ret->value->type) throw ParseException(std::string("bad type for ") + getExpressionName(ret) + ": " + printType(type) + " vs value type " + printType(ret->value->type), s.line, s.col);
```

Comment:
i have a vague memory this was needed for some spec test corner case... but since tests pass I guess either you fixed it, or I was wrong :)